yt_geotiff/utilities.py

```diff
@@ -127,4 +132,37 @@ def merge_dicts(*dict_args):
     for dictionary in dict_args:
         result.update(dictionary)
     return result
-
+      
+
+def rasterio_window_calc(selector):
+       """ 
+       This function reads information from either a sphere, box or region selector 
+       object and outputs the dimensions of a container: left edge, right edge,
+       width and height.
+       
+       Output uses:
+              - Construct dimensions to perform a Rasterio Window Read
+              - Provide a temporary update ro the image domain size             
+       """        
+       #pdb.set_trace()
+       if type(selector) == selector_shape.SphereSelector:
+              print('Sphere selector')
+              
+              selector_left_edge = [(selector.center[0] - selector.radius),(selector.center[1] - selector.radius), selector.center[2]]
+              selector_right_edge = [(selector.center[0] + selector.radius),(selector.center[1] + selector.radius), selector.center[2]] 
```

Comment:
This is just a bit more elegant:
```suggestion
              selector_left_edge = np.array(center)
              selector_left_edge[:2] -= selector.radius
              selector_right_edge = np.array(center)
              selector_right_edge[:2] += selector.radius
```